hazelcast/test/benchmark/README.md

```diff
@@ -0,0 +1,24 @@
+#Dependencies:
+    - [google benchmark](https://github.com/google/benchmark): Mac OS: brew install google-benchmark
+    - hazelcast-cpp-client 
+    - Boost
+
+#Build Instructions:
+mkdir build
+cd build
+cmake .. -DCMAKE_BUILD_TYPE=Debug -DCMAKE_PREFIX_PATH=/path/to/boost/install/dir;/path/to/hazelcast-cpp-client/install/dir
```

Comment:
Isn't `Release` more appropriate for benchmarks? `Debug` mode doesn't use optimization flags (`-O3`). 

From my computer: 

compiled with `-DCMAKE_BUILD_TYPE=Debug`:
```
--------------------------------------------------------------------------------
Benchmark                      Time             CPU   Iterations UserCounters...
--------------------------------------------------------------------------------
map_put/threads:32         50451 ns        33010 ns        22592 Put Rate=30.2934k/s
map_get/threads:32         51861 ns        31480 ns        22304 Get Rate=31.7657k/s
map_remove/threads:32      48031 ns        30989 ns        22304 Remove Rate=32.2697k/s
```

compiled with `-DCMAKE_BUILD_TYPE=Release`:
```
--------------------------------------------------------------------------------
Benchmark                      Time             CPU   Iterations UserCounters...
--------------------------------------------------------------------------------
map_put/threads:32         45438 ns        21636 ns        35392 Put Rate=46.2183k/s
map_get/threads:32         44832 ns        21132 ns        33024 Get Rate=47.3207k/s
map_remove/threads:32      44187 ns        20806 ns        32832 Remove Rate=48.0634k/s
```

Release is about ~50% better.

**Note:** In both of the above, the client itself was compiled in `Release` mode, the difference is just the benchmark suite.

Reply:
It is actually user specified and this was a recommendation for the default, but yes, I now changed it to Release as suggested in the README. By the way, even though they both use the Release version of hazelcast-cpp-client library it is also interesting to see this much difference for the performance test program build. This may be the affect of the templated library code optimization.

Reply:
Someone who is inexperienced with CMake (a QA person for example) may read this, its better to suggest the optimizing build for better comparison.

We have considerable amount of code in the header files, the optimization of them are done when the benchmark suite is compiled. The performance difference can be because of that.